app/uk/gov/hmrc/thirdpartyapplication/services/ApplicationService.scala

```diff
@@ -186,6 +187,16 @@ class ApplicationService @Inject()(applicationRepository: ApplicationRepository,
 
   }
 
+  def updateCidrBlocks(applicationId: UUID, newCidrBlocks: Set[String])(implicit hc: HeaderCarrier): Future[ApplicationData] = {
+    for {
+      validatedCidrBlocks <- fromTry(Try(newCidrBlocks.map(new SubnetUtils(_).getInfo.getCidrSignature))) recover {
+        case e: IllegalArgumentException => throw InvalidCidrBlockException(e.getMessage)
+      }
+      app <- fetchApp(applicationId)
+      updatedApp <- applicationRepository.save(app.copy(cidrBlocks = validatedCidrBlocks))
```

Comment:
We need to stop following this pattern of overwriting the entire document, as we'll lose data before too long (if we haven't already). Should be updating the collection of CIDR blocks within the document directly.

Reply:
Good point